### PR TITLE
fix: When killing a complex hive task, Hive MR of Yarn is not simultaneously killed, which results in task failuer. #843

### PR DESCRIPTION
**Pull request by HmhWz**
### What is the purpose of the change
(Fix: When killing a complex hive task, Hive MR of Yarn is not simultaneously killed, which results in task failuer. 
Related issues: #843 . )

### Does this pull request potentially affect one of the following parts:
- Dependencies (does it add or upgrade a dependency): (yes)
- Anything that affects deployment: (no)
- The MGS(Microservice Governance Services), i.e., Spring Cloud Gateway, OpenFeign, Eureka.: (no)

### Documentation
- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: linkis-engineconn-plugins/engineconn-plugins/hive/src/main/scala/com/webank/wedatasphere/linkis/engineplugin/hive/hook/HiveAddMetaTableNameHook.scala [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import com.webank.wedatasphere.linkis.engineconn.computation.executor.hook.ComputationExecutorHook
+import com.webank.wedatasphere.linkis.engineplugin.hive.exception.HiveQueryFailedException
+import org.apache.commons.lang.StringUtils
+import org.apache.hadoop.hive.conf.HiveConf.ConfVars
+
+import java.util
+import java.util.regex.Pattern
+import scala.collection.JavaConverters.mapAsScalaMapConverter
+
+/**
```
Comment (peacewong) [inline linkis-engineconn-plugins/engineconn-plugins/hive/src/main/scala/com/webank/wedatasphere/linkis/engineplugin/hive/hook/HiveAddMetaTableNameHook.scala:15]: Need to remove the author

File: linkis-engineconn-plugins/engineconn-plugins/hive/src/main/scala/com/webank/wedatasphere/linkis/engineplugin/hive/executor/HiveEngineConnExecutor.scala [hunk 2]
```diff
@@ -274,17 +150,150 @@ class HiveEngineConnExecutor(id: Int,
...
   }
 
 
   /**
-    * 在job完成之前，要将singleSqlProgressMap的剩余的内容全部变为成功
-    */
-  private def onComplete():Unit = {
-    if (engineExecutorContext != null){
-      val arrayBuffer:ArrayBuffer[JobProgressInfo] = new ArrayBuffer[JobProgressInfo]()
+   * 在job完成之前，要将singleSqlProgressMap的剩余的内容全部变为成功
```
Comment (peacewong) [inline linkis-engineconn-plugins/engineconn-plugins/hive/src/main/scala/com/webank/wedatasphere/linkis/engineplugin/hive/executor/HiveEngineConnExecutor.scala:292]: It is recommended to use English annotations

**Comment by HmhWz** [comment]
It has been improved and submitted




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2021年7月15日(星期四) 中午12:04
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [WeBankFinTech/Linkis] fix: When killing a complex hive task, Hive MR of Yarn is not simultaneously killed, which results in task failuer. #843 (#891)





 
@peacewong commented on this pull request.
 
 
In linkis-engineconn-plugins/engineconn-plugins/hive/src/main/scala/com/webank/wedatasphere/linkis/engineplugin/hive/hook/HiveAddMetaTableNameHook.scala:
 &gt; @@ -0,0 +1,57 @@ +package com.webank.wedatasphere.linkis.engineplugin.hive.hook + +import com.webank.wedatasphere.linkis.common.utils.{Logging, Utils} +import com.webank.wedatasphere.linkis.engineconn.common.creation.EngineCreationContext +import com.webank.wedatasphere.linkis.engineconn.computation.executor.execute.EngineExecutionContext +import com.webank.wedatasphere.linkis.engineconn.computation.executor.hook.ComputationExecutorHook +import com.webank.wedatasphere.linkis.engineplugin.hive.exception.HiveQueryFailedException +import org.apache.commons.lang.StringUtils +import org.apache.hadoop.hive.conf.HiveConf.ConfVars + +import java.util +import java.util.regex.Pattern +import scala.collection.JavaConverters.mapAsScalaMapConverter + +/**  
Need to remove the author
 
 
In linkis-engineconn-plugins/engineconn-plugins/hive/src/main/scala/com/webank/wedatasphere/linkis/engineplugin/hive/executor/HiveEngineConnExecutor.scala:
 &gt;      if (sql.trim.startsWith("create table") || sql.trim.startsWith("drop table")) true else false    }        /** -    * 在job完成之前，要将singleSqlProgressMap的剩余的内容全部变为成功 -    */ -  private def onComplete():Unit = { -    if (engineExecutorContext != null){ -      val arrayBuffer:ArrayBuffer[JobProgressInfo] = new ArrayBuffer[JobProgressInfo]() +   * 在job完成之前，要将singleSqlProgressMap的剩余的内容全部变为成功  
It is recommended to use English annotations

**Review by peacewong — APPROVED** [review]
LGTM.

**Pull request merged**